### PR TITLE
upgrade spring-boot to 3.1.0

### PR DESCRIPTION
**Pull request by RichardW98**
**What this PR does / why we need it**:
upgrade spring-boot to 2.7.13
**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story (FLPATH-xxxx)*:
Fixes #

**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Unit tests
- [ ] Integration tests
- [ ] CI
- [ ] Documentation
- [ ] Auto-generated SDK code

**Impacted services**
- [ ] Workflow Service
- [ ] Notification Service

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: workflow-service/src/main/java/com/redhat/parodos/security/SecurityConfiguration.java [hunk 1]
```diff
@@ -75,22 +75,28 @@ public HttpSecurity setHttpSecurity(HttpSecurity http) throws Exception {
                 .logout()
                 .logoutSuccessUrl("/login").permitAll();
         // @formatter:on
-		return http;
 	}
 
+	// @Bean
```
[the remark this answers is not in the document]
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/security/SecurityConfiguration.java:80]: yes it's redundant

**Review by masayag — COMMENTED** [review]
/lgtm

**Comment by masayag** [comment]
@RichardW98 so the plan will be to upgrade to spring 3, and if no issues - use it.
if there will be issues, then fallback to 2.7.latest that worked before this merge conflict.

File: workflow-service/src/main/java/com/redhat/parodos/security/SecurityConfiguration.java [hunk 2]
```diff
@@ -45,19 +44,20 @@
 @Configuration
 @Profile("!local")
 @DependsOn("org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor")
-public class SecurityConfiguration {
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
```
Comment (gabriel-farache) [inline workflow-service/src/main/java/com/redhat/parodos/security/SecurityConfiguration.java:47]: `WebSecurityConfigurerAdapter` is deprecated: `'org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter' is deprecated ` so we should not be using it
Instead we should use `AuthenticationManager`  'ie: https://docs.spring.io/spring-security/reference/servlet/authentication/passwords/ldap.html#servlet-authentication-ldap-pwd and https://spring.io/blog/2022/02/21/spring-security-without-the-websecurityconfigureradapter)

**Comment by RichardW98** [comment]
@masayag @gabriel-farache @eloycoto ,   I will keep AuthentionManagerBuilder here as we need it to read Embedded ldap server config from application.yml. after all Deprecated WebSecurityConfigurerAdapter is removed

**Comment by gabriel-farache** [comment]
/lgtm

And a question: so the v3 solved the issue with the autowired annotation for the configure method in the SecurityConfig?

**Comment by RichardW98** [comment]
> And a question: so the v3 solved the issue with the autowired annotation for the configure method in the SecurityConfig?

I think so,  the @Autowired  on method 'configure' will auto detect bean of AuthenticationManagerBuilder and apply it to the method

**Comment by RichardW98** [comment]
actually best practice now is to remove @Autowired and replace it with  private final local vars and constructor vars. let's fix it later if possible

**Pull request merged**